anndata/base.py

```diff
@@ -504,17 +395,21 @@ def varm(self):
     def var_names(self):
         return self.var.index
 
+    @property
+    def obs_names(self):
+        return self._adata.obs_names
+
     def __getitem__(self, index):
         oidx, vidx = self._normalize_indices(index)
         if self._adata is not None or not self._adata.isbacked: X = self._X[oidx, vidx]
         else: X = self._adata.file['raw.X'][oidx, vidx]
-        if isinstance(vidx, (int, np.int64)): vidx = slice(vidx, vidx+1, 1)
+        if isinstance(vidx, (int, np.int64)): 
+            vidx = slice(vidx, vidx+1, 1)  # why this?
```

Comment:
Better yet, I think I can replace it with an explanatory comment. This is a slice instead of an int in order to keep `adata[:, x: int].shape == adata[:, x: int].X.shape`, right?

Reply:
Yes, that's right!

PS: I think it's `adata[:, x:int]` and not `adata[:, x: int]`. See *slices* under https://www.python.org/dev/peps/pep-0008/#pet-peeves.

Reply:
Cool, I'll fix up that comment.

I'd meant `x: int` to read as a type annotation (`x` is an `int`), but now I'm seeing it's (at best) ambiguous in this context.

Reply:
Ah, yes, sure... I was already wondering... ;)